### PR TITLE
[SPARK-40478][DOCS] Add create datasource table options docs

### PR DESCRIPTION
**Pull request by ulysses-you**
### What changes were proposed in this pull request?

Add create datasource table options docs in `docs/sql-ref-syntax-ddl-create-table-datasource.md`.

### Why are the changes needed?

improve the missed docs and example.

### Does this PR introduce _any_ user-facing change?

yes, docs improved

### How was this patch tested?

N/A

### CODE REVIEW

File: docs/sql-ref-syntax-ddl-create-table-datasource.md [hunk 1]
```diff
@@ -117,6 +121,15 @@ CREATE TABLE student_copy USING CSV
 --Omit the USING clause, which uses the default data source (parquet by default)
 CREATE TABLE student (id INT, name STRING, age INT);
 
+--Use parquet data source with parquet storage options
+--The columns 'id' and 'name' enable the bloom filter during writing parquet file,
+--column 'age' does not enable
+CREATE TABLE student_parquet(id INT, name STRING, age INT) USING PARQUET
+    OPTIONS (
+      'parquet.bloom.filter.enabled'='true',
```
Comment (srowen) [inline docs/sql-ref-syntax-ddl-create-table-datasource.md:129]: Other examples of OPTIONS seem different, and would look more like:
`parquet.bloom.filter.enabled "true",`
It seems. Is this right and/or should we standardize?

Comment (ulysses-you) [reply docs/sql-ref-syntax-ddl-create-table-datasource.md:129]: yes, both of them are fine, see:
https://github.com/apache/spark/blob/59365703c51a36efca5390b7535489a839f51659/sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBaseParser.g4#L365-L395

I just follow the `TBLPROPERTIES` format in this docs.

**Comment by ulysses-you** [comment]
@srowen has other comments ? also cc @wangyum @cloud-fan

**Review by srowen — APPROVED** [review]

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**